File: ALGORITHM/Leetcode/java/007_Reverse_Integer.java
/**
 * @FileName :  007_Reverse_Integer.java
 * @Date : 2020. 9. 5.
 * @author : AoN
 * @Link : https://leetcode.com/problems/reverse-integer/
 * @Description : overflow를 어떻게 처리하는지가 핵심인 문제
 *
 */

 class Solution {
      public int reverse(int x) {
        long res = 0;
        while (x != 0) {
            res = res * 10 + x % 10;
            x = x / 10;
        }

        if (res < Integer.MIN_VALUE || res > Integer.MAX_VALUE) {
            return 0;
        } else {
            return (int)res;
        }
    }
}

File: ALGORITHM/Leetcode/python/007_Reverse_Integer.py
"""
 * @FileName : 007_Reverse_Integer.py
 * @Project : LeetCode
 * @Date : 2020. 9. 5.
 * @author : AoN
 * @Link : https://leetcode.com/problems/reverse-integer/
 * @Description :
 * 
"""


class Solution:
    def reverse(self, x: int) -> int:
        if x > 0:  # handle positive numbers
            ret = int(str(x)[::-1])
        if x <= 0:  # handle negative numbers
            ret = -1 * int(str(x * -1)[::-1])

        # handle 32 bit overflow
        min_val = -2 ** 31
        max_val = 2 ** 31 - 1

        return ret if ret < min_val or ret > max_val else 0